### PR TITLE
[rpc] Remove deprecated fields from getpeerinfo

### PR DESCRIPTION
**Pull request by amitiuttarwar**
This PR removes support for 3 fields on the `getpeerinfo` RPC that were deprecated in v0.21- `addnode`, `banscore` & `whitelisted`.

### CODE REVIEW

**Comment by amitiuttarwar** [comment]
These fields were deprecated in #19469, #19770, #19725. @jonatack & @luke-jr, if you get the chance, would appreciate your review :)

**Comment by amitiuttarwar** [comment]
Updated to include PR # in release notes

**Comment by amitiuttarwar** [comment]
I believe we can get rid of the `m_legacyWhitelisted` field on `CNode`, but am still trying to build confidence. 

There is a code comment: `//If this flag is present, the user probably expect that RPC and QT report it as whitelisted (backward compatibility)`. In this PR I'm removing the RPC endpoint, but I'm unable to find anywhere in QT that reveals this field. I'm unfamiliar with QT, maybe it was already removed, or maybe I'm missing it. 

@NicolasDorier - since you introduced this, wondering if you have any thoughts?

File: doc/release-notes.md [hunk 1]
```diff
@@ -63,6 +63,11 @@ P2P and network changes
 
 Updated RPCs
 ------------
+- `getpeerinfo` no longer supports the following fields: `addnode`, `banscore`,
+  and `whitelisted`, which were previously deprecated in 0.21. Instead of
+  `addnode`, the `connection_type` field will return manual. Instead of
```
Comment (jonatack) [inline doc/release-notes.md:68]: maybe s/no longer supports/no longer returns/

and s/will return manual/returns "manual"/

**Review by jonatack — COMMENTED** [review]
Approach ACK

File: doc/release-notes.md (hunk 1 above)
Comment (amitiuttarwar) [reply doc/release-notes.md]: sure, done.

**Comment by amitiuttarwar** [comment]
I realized this removes the last use of `m_legacyWhitelisted`, so I deleted the field.

**Comment by MarcoFalke** [comment]
Nice cleanup. Concept ACK

**Comment by amitiuttarwar** [comment]
rebased

**Comment by sipa** [comment]
utACK 454a4088a87eac5878070b26d13d5574da891877

File: src/net.cpp [hunk 2]
```diff
@@ -1123,8 +1121,6 @@ void CConnman::AcceptConnection(const ListenSocket& hListenSocket) {
     CNode* pnode = new CNode(id, nodeServices, GetBestHeight(), hSocket, addr, CalculateKeyedNetGroup(addr), nonce, addr_bind, "", ConnectionType::INBOUND, inbound_onion);
     pnode->AddRef();
     pnode->m_permissionFlags = permissionFlags;
-    // If this flag is present, the user probably expect that RPC and QT report it as whitelisted (backward compatibility)
-    pnode->m_legacyWhitelisted = legacyWhitelisted;
```
Comment (jnewbery) [inline src/net.cpp:1127]: You can get rid of the local variable `legacyWhitelisted` entirely.

**Review by jnewbery — COMMENTED** [review]
ACK 454a4088a87eac5878070b26d13d5574da891877.

You can remove a no-longer-needed local variable if you touch this again.

**Comment by michaelfolkson** [comment]
Looks like no one tested this before merging. Fixed in https://github.com/bitcoin/bitcoin/pull/20791 but as a post mortem should have been tested prior to merging.

**Comment by MarcoFalke** [comment]
> Looks like no one tested this before merging. Fixed in #20791 but as a post mortem should have been tested prior to merging.

I am not sure what you mean. clang doesn't emit this warning, as explained in https://github.com/bitcoin/bitcoin/pull/20791#issuecomment-751985730. Merging any pull will be impossible if maintainers are required to build with all supported compiler versions on all supported platforms.

Regardless, this has long been pointed out during review: https://github.com/bitcoin/bitcoin/pull/20755#pullrequestreview-558913726 with the comment: "You can remove a no-longer-needed local variable *if you touch this again*."

Moreover, `-Wunused` (and other warnings) are known to be broken for more than 13 years (https://gcc.gnu.org/bugzilla/show_bug.cgi?id=89180) in gcc. Rushing to "fix" (silence) them with patches that will break code (https://github.com/bitcoin/bitcoin/pull/20797#discussion_r549701244) is not the way to go. Personally, I prefer correct code over code that is incorrect, but doesn't emit any warnings.

**Comment by michaelfolkson** [comment]
This could have been easily caught with a simple manual grep right? For something basic like this someone should do that grep before it is merged? Or am I missing something?

The above is interesting context though, thanks.

**Comment by jonatack** [comment]
> Rushing to "fix" (silence) them with patches that will break code (https://github.com/bitcoin/bitcoin/pull/20797#discussion_r549701244) is not the way to go. Personally, I prefer correct code over code that is incorrect, but doesn't emit any warnings.

Diligence and vigilance does not mean I am suggesting we rush to break the code to silence the compiler. I'm not interested in blame games, only making bitcoin more robust.

**Comment by MarcoFalke** [comment]
> This could have been easily caught with a simple manual grep right?

Yes, and I assume that either grep or `--function-context` was used to find it in https://github.com/bitcoin/bitcoin/pull/20755#pullrequestreview-558913726.

@jonatack Sorry, didn't want to blame you. It was the most recent example that came to mind. There are many more examples, including many by me. I should have used one of myself.

**Comment by jonatack** [comment]
Thanks. I think given enough time it happens to everyone anyway. Teamwork makes the dream work.￼￼

**Comment by jnewbery** [comment]
> Looks like no one tested this before merging. Fixed in #20791 but as a post mortem should have been tested prior to merging.

This is an entirely unhelpful comment. The 'problem' here is that a variable is now unused and certain compilers warn about it. Unused variables are exactly that - unused - and so have no impact on the logic of the program. In fact, compilers will optimize them out, so I expect the produced object is exactly the same before and after #20791.

This was spotted in review, found to not be an issue, and would have been fixed if the branch was retouched. Instead it was fixed in a follow-up PR. There's absolutely no problem here.

> I assume that either grep or --function-context was used to find it in #20755 (review).

Just careful review. If a change means that a variable is no longer used in one place, it's good practice to see where else that variable is used.

**Comment by michaelfolkson** [comment]
It seems sloppy to me. Either remove it in the PR or merge it with an instruction that someone should open a new PR to remove it. I don't think it is great hygiene to just leave it hanging in a past comment (but maybe that is just me)

Don't want to belabor the point though (especially when being told it is "entirely unhelpful")

**Pull request merged**